config/config.go

```diff
@@ -0,0 +1,53 @@
+package config
+
+import "github.com/BurntSushi/toml"
+
+// Config encapsulates configurations of various Libnetwork components
+type Config struct {
+	Daemon    DaemonCfg
+	Cluster   ClusterCfg
+	Datastore DatastoreCfg
+}
+
+// DaemonCfg represents libnetwork core configuration
+type DaemonCfg struct {
+	Debug bool
+}
+
+// ClusterCfg represents cluster configuration
+type ClusterCfg struct {
+	Discovery string
+	Address   string
+	Heartbeat uint64
+}
+
+// DatastoreCfg represents Datastore configuration.
+// supports both embedded-server mode and client-only mode
+type DatastoreCfg struct {
+	Embedded bool
+	Client   DatastoreClientCfg
+	Server   DatastoreServerCfg
```

Comment:
Let's remove the datastore server config. We don't need it for non-embedded mode. For embedded mode either we don't need it or it may be different. So let's postpone this until we have embedded mode support